tests/integration/test_client_http.py

```diff
@@ -0,0 +1,203 @@
+import time
+import select
+import socket
+import random
+import threading
+from tests import unittest
+from contextlib import contextmanager
+
+import botocore.session
+from botocore.config import Config
+from botocore.vendored.six.moves import BaseHTTPServer, socketserver
+from botocore.exceptions import (
+    ConnectTimeoutError, ReadTimeoutError, EndpointConnectionError,
+    ConnectionClosedError,
+)
+
+
+class TestClientHTTPBehavior(unittest.TestCase):
+    def setUp(self):
+        self.port = random_port()
+        self.localhost = 'http://localhost:%s/' % self.port
+        self.session = botocore.session.get_session()
+
+    def test_can_proxy_https_request_with_auth(self):
```

Comment:
Do you think it's worth adding a test for plain http proxies?

Reply:
We could, but we would need a proxy server that understands it. I think at that point we're mostly testing urllib3, but it still could be worth it. I'll see what I can do.